10up/ElasticPress
# Delete indices of unused Indexables on a full sync

### Description of the Change

Unfortunately, this PR ended up doing more than I anticipated. The main goal of it is to delete indices of features that are no longer enabled. That will happen only if a full sync is performed.

To have that done, I had to differentiate registered and enabled Indexables and also check which indices exist before a sync is started. With that list of indices, it is now possible to send the mapping of an inexistent index, so this PR covers that too.

It also adds an optional `--status` parameter to the `get-indices` WP-CLI command. *Note:* This command outputs the list of indices related to the current site but they don't necessarily exist. The `get-cluster-indices` outputs the list of indices really present in the ES server.

As I had some problems running Cypress tests on WSL I've also added some instructions to fix those issues.

Closes #3186

### How to test the Change

1. Enable Comments and/or any other feature that creates an index
2. Run a full sync
3. See the new index created
4. Deactivate the feature(s)
5. Run a full sync
6. See the indices being deleted

Also
1. Delete any index of a currently enabled feature
2. Run a regular sync. See the `Index not present. Mapping sent` message.

### Changelog Entry
> Added - new `--status` parameter to the `get-indices` WP-CLI command.
> Changed - Indices of disabled features will be deleted during a full sync. Mappings of needed but unexistent indices will be added even during a regular sync.

### Credits

Props @felipeelia 

### Checklist:
